components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java

```diff
@@ -297,9 +297,7 @@ public Certificate getCertificate(String tenantDomain, InboundProtocol inboundPr
                     String.format(ErrorMessages.ERROR_CODE_INVALID_ARGUMENT.getDescription(), "Tenant domain"));
         }
         if (inboundProtocol == null) {
-            throw new IdentityKeyStoreResolverException(
-                    ErrorMessages.ERROR_CODE_INVALID_ARGUMENT.getCode(),
-                    String.format(ErrorMessages.ERROR_CODE_INVALID_ARGUMENT.getDescription(), "Inbound protocol"));
+            return getCertificate(tenantDomain);
```

Comment:
As per the comments there is no need to throw client error, we can fallback to default by getting certificate of the tenant domain without bound to the protocol